src/binding.ts

```diff
@@ -63,13 +64,13 @@ export class BindingService {
         const connectionId = binding.connectionId || binding.serverId || '<default>';
         let connectionBindings = bindings.get(connectionId);
         if (!bindings.has(connectionId)) {
```

Comment:
I think these two lines are useless? The `bindings` will always be empty here, as set on line 58.

Reply:
It's a lazy initialization: initially, `bindings` is empty, and we are lazily creating a new `Map` for each `connectionId` encountered while iterating over the workspace folders (and same below, lazily creating an array to hold bindings for a given project the first time we find one folder bound to it).

I can maybe rename the variables to make it more explicit.

Reply:
Thanks for the clarification! I missed the for loop :)